app/src/renderer/components/network/PageBlock.vue

```diff
@@ -0,0 +1,154 @@
+            <dl class="info_dl colored_dl">
+              <dt>Transactions</dt>
+              <dd>{{ block.block.data.txs || `No Transactions` }}</dd>
+            </dl>
+          </div>
+        </div>
+      </div>
+
+      <div class="page-profile__section block">
+        <div class="row">
+          <div class="column">
+            <dl class="info_dl colored_dl">
+              <dt>Evidence</dt>
+              <dd>{{ block.block.evidence.evidence || `No Evidence` }}</dd>
+            </dl>
+          </div>
+        </div>
+      </div>
+
+      <div class="page-profile__section block">
+        <div class="row">
+          <div class="column">
+            <dl class="info_dl colored_dl">
+              <dt>Pre Commits</dt>
+              <dd v-if="!block.block.last_commit.precommits">No precommits</dd>
+              <table v-else class="pre-commits data-table">
+                <thead>
+                  <panel-sort :properties="properties" />
+                </thead>
+                <tbody>
+                  <tr
+                    v-for="(precommit, index) in block.block.last_commit
+                      .precommits"
+                    :key="index"
+                    class="block data-table__row"
+                  >
+                    <td>{{ precommit.validator_address }}</td>
+                    <td>
+                      {{
+                        moment(precommit.timestamp).format(
+                          "MMM Do YYYY, HH:mm:ss"
+                        )
+                      }}
+                    </td>
+                    <td>{{ precommit.round }}</td>
+                  </tr>
+                </tbody>
+              </table>
+            </dl>
+          </div>
+        </div>
+      </div>
+    </template>
+  </tm-page>
+</template>
+
+<script>
+import moment from "moment"
+import axios from "axios"
+import { mapGetters } from "vuex"
+import num from "scripts/num"
+import PanelSort from "staking/PanelSort"
+import ToolBar from "common/ToolBar"
+import TmBalance from "common/TmBalance"
+import TmDataError from "common/TmDataError"
+import TmPage from "common/TmPage"
+export default {
+  name: `page-block`,
+  components: {
+    PanelSort,
+    TmBalance,
+    ToolBar,
+    TmDataError,
+    TmPage
+  },
+  data: () => ({
+    num,
+    moment,
+    header: null,
+    block: null
+  }),
+  computed: {
+    ...mapGetters([`connected`, `blocks`]),
+    properties() {
+      return [
+        {
+          title: `Proposer`
+        },
+        {
+          title: `Time`
+        },
+        {
+          title: `Round`
+        }
+      ]
+    }
+  },
+  watch: {
+    $route: `getBlock`
+  },
+  async mounted() {
+    await this.getBlock()
+  },
+  methods: {
+    async getBlock() {
+      const url =
+        `http://localhost:8080/` + `blocks/` + this.$route.params.height
+      const block = await axios.get(url)
+      this.header = block.data.block.header
+      this.block = block.data
```

Comment:
I think we may want to move this logic away of the UX, it should stay into an action in `blocks.js`, that could be dispached form here and it returns the block (optionally we could add it to the state but I think it's fine to fetch it again OR we have to build up a dict of already fetched blocks)

Reply:
let's check if `queryBlockInfo` suits your needs

Reply:
yep. great idea. i've refactored this.